flax/linen/transforms.py

```diff
@@ -446,9 +486,15 @@ def vmap(target: Target,
     axis_name: Specifies a name for the batch axis. Can be used together
       with parallel reduction primitives (e.g. `jax.lax.pmean`,
       `jax.lax.ppermute`, etc.)
+    methods: If `target` is a `Module`, the methods of `Module` to vmap over.
+
+  Returns:
+    A batched/vectorized version of ``target``, with the same arguments but with
+    extra axes at positions indicated by ``in_axes``, and the same return value,
+    but with extra axes at positions indicated by ``out_axes``.
   """
   return lift_transform(
-      lift.vmap, target, variable_axes, split_rngs,
+      lift.vmap, target, _if_none(variable_axes, {}), _if_none(split_rngs, {}),
```

Comment:
It would be more consistent to do this in lift.py.
I think the linter is wrong here btw: mutating the global default is just as bad as mutating the users input unexpectedly. That's why the annotating is Mapping not MutableMapping such that updating the dicts in-place is a type error

Reply:
The reason why a Mapping is still not allowed to have a mutable global default is that it could get passed to code that does not do any type checking (so it may treat it as a regular dict and mutate it anyway): https://google.github.io/styleguide/pyguide.html#2124-decision

Reply:
But that also happens if a user passes in a mapping that they want to reuse, right? I think the pragmatic solution is to make the default a non-mutable Mapping like FrozenDict({}) which acts as a runtime test for the lack of mutation and you don't need the redundant None checks

Reply:
I agree, that seems like the nicest solution. Updated it.